eval.c

```diff
@@ -1079,11 +1079,21 @@ rb_mod_prepend(int argc, VALUE *argv, VALUE module)
 {
     int i;
     ID id_prepend_features, id_prepended;
+    VALUE new_mod;
 
     CONST_ID(id_prepend_features, "prepend_features");
     CONST_ID(id_prepended, "prepended");
 
-    rb_check_arity(argc, 1, UNLIMITED_ARGUMENTS);
+    if(rb_block_given_p()) {
```

Comment:
@nobu My assumption in this case is that if a block is given then no arguments are required, which would then result in an arity check of `0, UNLIMITED_ARGUMENTS` which as I understand it would be a noop. Is that not the case?